server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java

```diff
@@ -685,22 +690,46 @@ private BoolQueryBuilder createBoolFilter(IssueQuery query) {
-            .filter(rangeQuery(IssueIndexDefinition.FIELD_ISSUE_FUNC_CREATED_AT).gte(epochMillisToEpochSeconds(from)))
-          )
+            .filter(rangeQuery(IssueIndexDefinition.FIELD_ISSUE_FUNC_CREATED_AT).gte(epochMillisToEpochSeconds(from))))
           .subAggregation(AggregationBuilders.count(projectUuid + "_count").field(IssueIndexDefinition.FIELD_ISSUE_KEY))
-          .subAggregation(AggregationBuilders.max(projectUuid + "_maxFuncCreatedAt").field(IssueIndexDefinition.FIELD_ISSUE_FUNC_CREATED_AT))
-        );
+          .subAggregation(AggregationBuilders.max(projectUuid + "_maxFuncCreatedAt").field(IssueIndexDefinition.FIELD_ISSUE_FUNC_CREATED_AT)));
     });
     SearchResponse response = request.get();
     return response.getAggregations().asList().stream()
       .map(x -> (InternalFilter) x)
       .flatMap(projectBucket -> {
-      long count = ((InternalValueCount) projectBucket.getAggregations().get(projectBucket.getName() + "_count")).getValue();
-      if (count < 1L) {
-        return Stream.empty();
-      }
-      long lastIssueDate = (long) ((InternalMax) projectBucket.getAggregations().get(projectBucket.getName() + "_maxFuncCreatedAt")).getValue();
-      return Stream.of(new ProjectStatistics(projectBucket.getName(), count, lastIssueDate));
-    }).collect(MoreCollectors.toList(projectUuids.size()));
+        long count = ((InternalValueCount) projectBucket.getAggregations().get(projectBucket.getName() + "_count")).getValue();
+        if (count < 1L) {
+          return Stream.empty();
+        }
+        long lastIssueDate = (long) ((InternalMax) projectBucket.getAggregations().get(projectBucket.getName() + "_maxFuncCreatedAt")).getValue();
+        return Stream.of(new ProjectStatistics(projectBucket.getName(), count, lastIssueDate));
+      }).collect(MoreCollectors.toList(projectUuids.size()));
   }
+
+  public List<BranchStatistics> searchBranchStatistics(List<String> branchUuids) {
+    if (branchUuids.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    SearchRequestBuilder request = client.prepareSearch(IssueIndexDefinition.INDEX_TYPE_ISSUE)
+      .setQuery(
+        boolQuery()
+          .must(termsQuery(IssueIndexDefinition.FIELD_ISSUE_BRANCH_UUID, branchUuids))
+          .must(termQuery(IssueIndexDefinition.FIELD_ISSUE_IS_MAIN_BRANCH, Boolean.toString(false))))
+      .setSize(0)
+      .addAggregation(AggregationBuilders.terms("branchUuids")
+        .field(IssueIndexDefinition.FIELD_ISSUE_BRANCH_UUID)
+        .size(branchUuids.size())
+        .subAggregation(AggregationBuilders.terms("types")
+          .field(IssueIndexDefinition.FIELD_ISSUE_TYPE)));
+    SearchResponse response = request.get();
+    return ((StringTerms) response.getAggregations().get("branchUuids")).getBuckets().stream()
+      .map(bucket -> new BranchStatistics(bucket.getKeyAsString(),
+        ((StringTerms) bucket.getAggregations().get("types")).getBuckets()
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "StringTerms". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1905)